lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -732,66 +733,157 @@ class MetricsEndpointProvider(Object):
                 `self` in the instantiating class.
-            name: a string name of the relation between `charm` and
-                the Prometheus charmed service.
-            service_event: a `CharmEvent` in response to which each charm unit
-                must advertise its scrape endpoint host address.
+            relation_name: an optional string name of the relation between `charm`
+                and the Prometheus charmed service. You will need to specify this
+                parameter only if your charm provides more than one relation with
+                `prometheus_scrape` as interface, which is very uncommon: if your
+                charm has multiple metrics endpoints, you can expose them over
+                one relation as separate entries in the `jobs` list.
+                If `relation_name` is not specified, the
+                `MetricsEndpointProvider` will look for the existence of precisely
+                one relation provided by the charm and using the `prometheus_scrape`
+                interface and, if none or multiple are found, it will raise a
+                ops.model.ModelError exception.
             jobs: an optional list of dictionaries where each
                 dictionary represents the Prometheus scrape
                 configuration for a single job. When not provided, a
                 default scrape configuration is provided for the
-                `/metrics` endpoint pooling using port `80`.
+                `/metrics` endpoint pooling all units of this charm on port `80`.
             alert_rules_path: an optional path for the location of alert rules
-                files.  Defaults to "src/prometheus_alert_rules" at the top level
-                of the charm repository.
+                files.  Defaults to "./prometheus_alert_rules",
+                resolved from the directory hosting the charm entry file.
+                The alert rules are automatically update on charm upgrade.
         """
-        super().__init__(charm, name)
+        if not relation_name:
+            # Check if there is just one relation with the right interface
+            prometheus_scrape_relations = [
+                relation_name
+                for relation_name, relation_meta in charm.meta.provides.items()
+                if relation_meta.interface_name == "prometheus_scrape"
+            ]
+
+            if len(prometheus_scrape_relations) == 1:
+                relation_name = prometheus_scrape_relations[0]
+            elif len(prometheus_scrape_relations) == 0:
+                raise ModelError(
+                    "No provided relation with the 'prometheus_scrape' interface found"
+                )
+            else:
+                raise ModelError(
+                    "Multiple relations with the 'prometheus_scrape' interface found: "
+                    f" {prometheus_scrape_relations}"
+                )
+
+        if not alert_rules_path:
```

Comment:
Similarly extracting this logic of determining alert rules path into a separate function would be good practice.

Reply:
See `def get_this_script_dir() -> Path:`
https://github.com/canonical/lma-light-bundle/blob/c72b9a02adec54ac67c01642552af00d77831902/tests/integration/test_bundle.py#L35